module-4/backend/week_one.md

```diff
@@ -1,20 +1,41 @@
-## Week One - Module 4 Recap
-
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
-
-Note: When you're done, submit a PR. 
+Week One, Mod 4 Checks for Understanding
 
 1. What's the most useful thing you learned from completing the intermission week work?
+  The different options for debugging and 'prying' into javascript.  I use the console, node, and 'pryjs' regularly to try and figure out what exactly my code is doing.
+
 2. What are some tools to help debug JavaScript code?
+  debugger, pryjs, the console, and the node console in your terminal window
+
 3. What are some tools you need in order to unit test your JavaScript?
+  You need to make sure you have installed mocha and chai.  You also need to make sure you export the file containing the functions you are testing.  I also find installing pry particularly useful for figuring out how/why tests are not passing.
+
 4. What is the syntax for invoking a function?
-5. What's the difference between `==` and `===` in JavaScript?
-6. What's the difference between asynchronous and synchronous JavaScript? 
+  To invoke a function you type the function's name followed by ().  If the function takes arguments, the argument(s) go inside the parenthesis.
+
+5. What's the difference between == and === in JavaScript?
+  == is the 'abstract equality comparison operator'.  This will attempt to resolve the datatypes via type conversion being compared before comparing. So, console.log(1 == '1') is true.  The string and integer are the equal because both are one.
+  === is the 'strict equality comparison operator'.  This will not try to resolve the datatype and will compare literally.  So, console.log(1 === '1') is false, because one datatype is a string and the other is an integer.
+
+6. What's the difference between asynchronous and synchronous JavaScript?
+  Synchronous javascript runs code one thing after the next, or in order.
+  Asynchronous is not completed in any specific order; one thing does not necessarily wait for the thing before it to finish before executing.
```

Comment:
Remember that it still runs in order, it will just move on to the next line if a function is asynchronous instead of waiting for each line to evaluate before moving onto the next.